lib/ansible/modules/cloud/azure/azure_rm_cosmosdbaccount.py

```diff
@@ -0,0 +1,544 @@
+                       for this value is 1 - 2,147,483,647. Required when I(default_consistency_policy) is set to C(bounded_staleness)."
+                type: int
+            max_interval_in_seconds:
+                description:
+                    - "When used with the Bounded Staleness consistency level, this value represents the time amount of staleness (in seconds) tolerated.
+                       Accepted range for this value is 5 - 86400. Required when I(default_consistency_policy) is set to C(bounded_staleness)."
+                type: int
+    geo_rep_locations:
+        description:
+            - An array that contains the georeplication locations enabled for the Cosmos DB account.
+            - Required when C(state) is I(present).
+        type: list
+        suboptions:
+            name:
+                description:
+                    - The name of the region.
+            failover_priority:
+                description:
+                    - "The failover priority of the region. A failover priority of 0 indicates a write region. The maximum value for a failover priority =
+                       (total number of regions - 1). Failover priority values must be unique for each of the regions in which the database account exists."
+                type: int
+    database_account_offer_type:
+        description:
+            - Database account offer type, for example I(Standard)
+            - Required when C(state) is I(present).
+    ip_range_filter:
+        description:
+            - "Cosmos DB Firewall Support: This value specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list
+               of client IPs for a given database account. IP addresses/ranges must be comma separated and must not contain any spaces."
+    is_virtual_network_filter_enabled:
+        description:
+            - Flag to indicate whether to enable/disable Virtual Network ACL rules.
+    enable_automatic_failover:
+        description:
+            - "Enables automatic failover of the write region in the rare event that the region is unavailable due to an outage. Automatic failover will
+               result in a new write region for the account and is chosen based on the failover priorities configured for the account."
+        type: bool
+    capabilities:
+        description:
+            - List of Cosmos DB capabilities for the account
+        type: list
+        suboptions:
+            name:
+                description:
+                    - "Name of the Cosmos DB capability. For example I(EnableCassandra), I(EnableTable) or I(EnableGremlin)"
+    virtual_network_rules:
+        description:
+            - List of Virtual Network ACL rules configured for the Cosmos DB account.
+        type: list
+        suboptions:
+            id:
+                description:
```

Comment:
I'm loathe to add another layer of depth to this, but should we consider having this follow the same patterns the other modules are now with taking either a flat ID or a name + optional RG name?